Src/Our.Umbraco.Photon/Web/UI/App_Plugins/Photon/Js/photon.js

```diff
@@ -3,66 +3,85 @@
+            console.log($scope.model.value);
 
-                //clear the ui
-                $scope.imageSrc = undefined;
-                if ($scope.model.value) {
-                    delete $scope.model.value;
-                }
+            $scope.tmpModel = $scope.model.value || angular.copy(defaultValue);
+            $scope.tmpModel.imageSrc = "";
+
+            if ($scope.model.value && $scope.model.value.imageId > 0) {
+                entityResource.getById($scope.model.value.imageId, "Media").then(function (media) {
+
+                    // Extract media url
+                    var src = media.hasOwnProperty("metaData")
+                        ? media.metaData["umbracoFile"] : "";
+
+                    // Check for image cropper
+                    if (typeof src === 'object' && src.hasOwnProperty("PropertyEditorAlias") && src["PropertyEditorAlias"] == "Umbraco.ImageCropper") {
+                        src = src.Value.src;
+                    }
+
+                    // Src is some other object so set to empty string
+                    if (typeof src === 'object') {
+                        src = "";
+                    }
+
+                    // Update scope
+                    $scope.tmpModel.imageSrc = src;
+                });
             }
-        };
+        }
 
-        $scope.done = function () {
-            $scope.currentTag = undefined;
+        setupViewModel();
+
+        $scope.showAdd = function () {
+            return !$scope.tmpModel.imageSrc;
         };
 
-        // on image selected, update the img tag
-        $scope.$on("filesSelected", function (ev, args) {
+        $scope.add = function () {
 
-            $scope.model.value = $.extend({}, defaultValue);
+            $scope.mediaPickerOverlay = {
+                view: "mediapicker",
+                title: "Select media",
+                multiPicker: false,
+                show: true,
+                submit: function (model) {
 
-            if (args.files && args.files[0]) {
-                fileManager.setFiles($scope.model.alias, args.files);
+                    if (model.selectedImages.length > 0 && model.selectedImages[0].hasOwnProperty("image")) {
+                        $scope.tmpModel.imageId = model.selectedImages[0].id;
+                        $scope.tmpModel.imageSrc = model.selectedImages[0].image;
+                    }
+                    
+                    $scope.mediaPickerOverlay.show = false;
+                    $scope.mediaPickerOverlay = null;
 
-                var reader = new FileReader();
-                reader.onload = function (e) {
-                    $scope.$apply(function () {
-                        $scope.imageSrc = e.target.result;
-                    });
-                };
+                }
+            };
 
-                reader.readAsDataURL(args.files[0]);
+        };
+
+        $scope.clear = function () {
+            if (confirm("You sure?")) {
```

Comment:
You could use this language key here: https://github.com/umbraco/Umbraco-CMS/blob/dev-v7/src/Umbraco.Web.UI/umbraco/config/lang/en.xml#L220